packages/repository/src/decorators/model.decorator.ts

```diff
@@ -32,7 +32,7 @@ export const MODEL_WITH_PROPERTIES_KEY = MetadataAccessor.create<
   ClassDecorator
 >('loopback:model-and-properties');
 
-export type PropertyMap = MetadataMap<PropertyDefinition>;
+export type PropertyMap = MetadataMap<Partial<PropertyDefinition>>;
```

Comment:
This change seems a bit unrelated to the test being added, I am also not sure if is a good idea to allow only partial property definitions in `PropertyMap`. Shouldn't we (the `@model` decorator?) guarantee that `PropertyMap` always contains valid (full) property definitions only?

Reply:
only "type" is required (but is inferred when is not provided)

https://github.com/strongloop/loopback-next/blob/11267618b58d437cd9a3c28f4c2f4f48cfe930c0/packages/repository/src/decorators/model.decorator.ts#L87-L89

here designType does not provide a properly type (string | Function | undefined) instead provide (string | Function) because `propDef.type` is not nullable, when in fact it is (that's why  we check and throw CANNOT_INFER_PROPERTY_TYPE error). i'm open to use another approach thought. like only change
https://github.com/strongloop/loopback-next/blob/11267618b58d437cd9a3c28f4c2f4f48cfe930c0/packages/repository/src/decorators/model.decorator.ts#L83

Reply:
I though that `PropertyMap` is being used by `ModelDefinition` too. I did a [quick search](https://github.com/strongloop/loopback-next/search?q=PropertyMap) and see that this type is used only for decorator-related metadata, in which case this change is fine with me. (I was worrying about changing ModelDefinition contract, which is not the case.)